src/fs/metadata_log.cc

```diff
@@ -134,40 +168,54 @@ void metadata_log::memory_only_delete_inode(inode_t inode) {
     assert(not it->second.is_linked());
 
     std::visit(overloaded {
-        [](const inode_info::directory& dir) {
+        [](inode_info::directory& dir) {
             assert(dir.entries.empty());
         },
-        [](const inode_info::file&) {
-            // TODO: for compaction: update used inode_data_vec
+        [&](inode_info::file& file) {
+            cut_out_data_range(file, {
+                0,
+                std::numeric_limits<decltype(file_range::end)>::max()
+            });
+            ondisk_entry_size(ondisk_truncate{});
         }
     }, it->second.contents);
 
     _inodes.erase(it);
+    _ondisk_log_size += ondisk_entry_size(ondisk_delete_inode{});
+    _rewrite_log_size -= ondisk_entry_size(ondisk_create_inode{});
 }
 
 void metadata_log::memory_only_small_write(inode_t inode, file_offset_t file_offset, temporary_buffer<uint8_t> data) {
     inode_data_vec data_vec = {
         {file_offset, file_offset + data.size()},
         inode_data_vec::in_mem_data {std::move(data)}
     };
-
+    size_t log_increase = ondisk_entry_size(ondisk_small_write_header{0,0,static_cast<uint16_t>(data.size()),0});
```

Comment:
`ondisk_entry_size(ondisk_small_write_header{0,0,static_cast<uint16_t>(data.size()),0});` doesn't look good. Try to simplify those kinds of calls. Maybe some set of functions like `ondisk_entry_size<ondisk_small_write_header>(data.size());` would be better?

Reply:
since we call this several times, it indeed makes sense to have a helper function